src/ArrayOfArraysView.hpp

```diff
@@ -734,6 +734,40 @@ class ArrayOfArraysView
     }, m_values, buffers ... );
   }
 
+  /**
+   * @brief Clears the array and creates a new array with the given number of sub-arrays.
+   * @param numSubArrays The new number of arrays.
+   * @param offsets A pointer to an array of length @p numSubArrays+1 containing the offset
+   *   of each new sub array. Offsets are precomputed by the caller.
+   * @param buffers A variadic pack of buffers to treat similarly to m_values.
+   */
+  template< typename ... BUFFERS >
+  void resizeFromOffsets( INDEX_TYPE const numSubArrays,
+                          INDEX_TYPE const * const offsets,
+                          BUFFERS & ... buffers )
+  {
+    LVARRAY_ASSERT( arrayManipulation::isPositive( numSubArrays ) );
+    LVARRAY_ASSERT_EQ( offsets[0], 0 );
+    LVARRAY_ASSERT( sortedArrayManipulation::isSorted( offsets, offsets + numSubArrays + 1 ) );
+
+    destroyValues( 0, m_numArrays, buffers ... );
+
+    bufferManipulation::reserve( m_sizes, m_numArrays, MemorySpace::host, numSubArrays );
+    std::fill_n( m_sizes.data(), numSubArrays, 0 );
```

Comment:
I know this is just copied from above but maybe this should use `umpireInterface::memset`. I don't think it's important at the moment but if eventually we want to be able to do this without allocating anything on host it'll be necessary.

Reply:
Ok, done